tests/core/interceptor.c

```diff
@@ -817,3 +818,37 @@ replacement_malloc (gsize size)
 
   return GSIZE_TO_POINTER (size);
 }
+
+TESTCASE (replace_keep_original)
+{
+  gpointer (* malloc_impl) (gsize size);
+  gpointer (* original_impl) (gsize size) = NULL;
+  guint counter = 0;
+  volatile gpointer ret;
+
+  if (RUNNING_ON_VALGRIND)
+  {
+    g_print ("<skipping, not compatible with Valgrind> ");
+    return;
+  }
+
+  malloc_impl = interceptor_fixture_get_libc_malloc ();
+
+  g_assert_cmpint (gum_interceptor_replace (fixture->interceptor, malloc_impl,
+      replacement_malloc, &counter, (void **)&original_impl), ==, GUM_REPLACE_OK);
+  g_assert_nonnull(original_impl);
```

Comment:
```suggestion
  g_assert_nonnull (original_impl);
```

(There's a style guide [here](https://github.com/frida/frida/blob/main/CONTRIBUTING.md) by the way.)